prow/plugins/verify-owners/verify-owners.go

```diff
@@ -106,22 +125,72 @@ func handlePullRequest(pc plugins.Agent, pre github.PullRequestEvent) error {
 			break
 		}
 	}
-	return handle(pc.GitHubClient, pc.GitClient, pc.OwnersClient, pc.Logger, &pre, pc.PluginConfig.Owners.LabelsBlackList, pc.PluginConfig.TriggerFor(pre.Repo.Owner.Login, pre.Repo.Name), skipTrustedUserCheck, cp)
+
+	prState := state{
+		org:          pre.Repo.Owner.Login,
+		repo:         pre.Repo.Name,
+		repoFullName: pre.Repo.FullName,
+		number:       pre.Number,
+	}
+
+	return handle(pc.GitHubClient, pc.GitClient, pc.OwnersClient, pc.Logger, &pre.PullRequest, prState, pc.PluginConfig.Owners.LabelsBlackList, pc.PluginConfig.TriggerFor(pre.Repo.Owner.Login, pre.Repo.Name), skipTrustedUserCheck, cp)
+}
+
+func handleGenericCommentEvent(pc plugins.Agent, e github.GenericCommentEvent) error {
+	cp, err := pc.CommentPruner()
+	if err != nil {
+		return err
+	}
+
+	var skipTrustedUserCheck bool
+	for _, r := range pc.PluginConfig.Owners.SkipCollaborators {
+		if r == e.Repo.Name {
+			skipTrustedUserCheck = true
+			break
+		}
+	}
+
+	return handleGenericComment(pc.GitHubClient, pc.GitClient, pc.OwnersClient, pc.Logger, &e, pc.PluginConfig.Owners.LabelsBlackList, pc.PluginConfig.TriggerFor(e.Repo.Owner.Login, e.Repo.Name), skipTrustedUserCheck, cp)
+}
+
+func handleGenericComment(ghc githubClient, gc *git.Client, roc repoownersClient, log *logrus.Entry, ce *github.GenericCommentEvent, labelsBlackList []string, triggerConfig plugins.Trigger, skipTrustedUserCheck bool, cp commentPruner) error {
+	// Only consider open PRs and new comments.
+	if ce.IssueState != "open" && !ce.IsPR && ce.Action != github.GenericCommentActionCreated {
```

Comment:
I think these should be `||`ed, each of them separately looks like a valid condition to bail out.

Reply:
Fixed! And these were not caught by tests because of a bad copy/paste. :woman_facepalming: